client/domain/ReinforcementObjectFactory.js

```diff
@@ -0,0 +1,120 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating new frontend instances of Reinforcement
+ * domain objects.
+ */
+
+tie.factory('ReinforcementObjectFactory', [
+  'ReinforcementBulletObjectFactory', function(
+      ReinforcementBulletObjectFactory) {
+    /**
+    * The reinforcement object contains the current state of reinforcement
+    * comments about a particular task in a question.
+    * Fields:
+    *  task: The corresponding task for the reinforcement comments.
+    *  passedTags: Dictionary containing tags (test sets) that the code
+    *              passed/failed.
```

Comment:
Add: A tag is considered "failed" if at least one of the correctness tests corresponding to the tag have failed.